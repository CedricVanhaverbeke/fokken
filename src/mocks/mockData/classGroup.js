import faker from 'faker';

export const classGroupDetails = {
  name: '1A',
};

export const classGroupLesson = {
  id: 'd49515ed-972c-11e9-b935-0a0034000161',
  publishStatus: 'OPEN_FOR_HOME',
  updatedAt: '2019-01-01T00:00:00.000+0100',
  resultsPublished: false,
  givenOn: null,
  givenBy: null,
};

export const classGroupStudents = Array.from(Array(20), (_, i) => ({
  id: i,
  username: faker.internet.userName(),
  firstName: faker.name.firstName(),
  lastName: faker.name.lastName(),
  lastLogin: faker.date.past().toISOString(),
  idp: 'FTRPRF',
  role: 'SCHOOLSTUDENT',
  schoolId: faker.random.uuid(),
}));
