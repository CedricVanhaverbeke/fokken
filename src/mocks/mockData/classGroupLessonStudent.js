import faker from 'faker';

export default Array.from(Array(20), (_, i) => ({
  id: i,
  submittedAt: Math.random() < 0.2 ? null : faker.date.past().toISOString(),
}));
