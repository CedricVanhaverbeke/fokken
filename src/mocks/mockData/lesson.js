import faker from 'faker';

export const lessonDetails = {
  id: 'd49515ed-972c-11e9-b935-0a0029000023',
  lessonContentId: 955,
  title: '(a)Sociale media',
  defaultPublishStatus: 'CLOSED',
  sequence: 0,
  document: null,
};

const generateLessonQuestions = () => {
  faker.seed(69);

  return {
    archived: faker.random.boolean(),
    centurySkills: [],
    id: 1000,
    published: faker.random.boolean(),
    slides: Array.from(Array(10), (_, i) => ({
      content: faker.lorem.paragraph(),
      id: i,
      lessonId: 1000,
      module: 'NORMAL',
      question: {
        id: 100 + i,
        questionAnswers: Array.from(Array(4), (_, i) => ({
          id: 200 + i,
          track: true,
          value: faker.lorem.sentence(),
          correct: faker.random.boolean(),
        })),
      },
    })),
  };
};

const generateLessonAnswers = () => {
  faker.seed(69);

  return Array.from(Array(10), (_, i) => ({
    id: faker.random.uuid(),
    lessonContentId: faker.random.uuid(),
    answerType: 'MULTIPLECHOICE',
    questionId: 100 + i,
    userId: faker.random.uuid(),
    answers: Array.from(Array(2), () => ({
      id: faker.random.uuid(),
      multipleChoiceAnswerId: 200 + (Math.ceil(Math.random() * 10) % 4),
    })),
    submittedAt: faker.date.past().toISOString(),
  }));
};

export const lessonAnswers = generateLessonAnswers();
export const lessonQuestions = generateLessonQuestions();
