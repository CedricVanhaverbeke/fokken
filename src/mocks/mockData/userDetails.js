import faker from 'faker';

export default {
  username: faker.internet.userName(),
  firstName: faker.name.firstName(),
  lastName: faker.name.lastName(),
};
