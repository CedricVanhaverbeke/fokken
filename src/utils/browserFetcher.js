var cookie = require('cookie');

const browserFetcher = (resource, init) => {
  const { authorization: token } = cookie.parse(document.cookie);

  return fetch(resource, {
    ...init,
    authorization: `Bearer ${token}`,
  }).then((res) => res.json());
};

export default browserFetcher;
