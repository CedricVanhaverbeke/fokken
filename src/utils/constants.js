export const EMPTY_PROPS = {
  props: {},
};
