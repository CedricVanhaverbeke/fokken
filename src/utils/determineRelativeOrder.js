const determineRelativeOrder = (order, ownIndex) => {
  const relativeOrder = Array(order.length).fill(0);
  order.forEach((entry, i) => {
    const newIndex =
      i === ownIndex
        ? 0
        : i <= ownIndex
        ? order.length - ownIndex + i
        : i - ownIndex;

    relativeOrder[newIndex] = entry;
  });

  return relativeOrder;
};

export default determineRelativeOrder;
