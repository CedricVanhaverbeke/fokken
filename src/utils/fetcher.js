import cookie from 'cookie';
// eslint-disable-next-line no-unused-vars
import next from 'next';

export const defaultFetcher = (endpoint, token) => {
  return fetch(endpoint, {
    headers: {
      Authorization: `Bearer ${token}`,
    },
  }).then((res) => res.json());
};

export const browserFetcher = (endpoint) => {
  const { authorization: token } = cookie.parse(document.cookie);

  return defaultFetcher(endpoint, token);
};
