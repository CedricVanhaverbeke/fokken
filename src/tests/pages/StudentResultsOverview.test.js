import { renderPage } from '../utils';

import { classGroupLessonStudents } from '@/mocks/mockData/classGroup';

describe('StudentResultsOverview', () => {
  const students = classGroupLessonStudents;

  it('should render the results page when navigating to the URL', async () => {
    const { container, getByText } = await renderPage(
      '/classgroups/23234/lessons/123',
    );
    expect(container).toMatchSnapshot();
    students.forEach(({ firstName, lastName }) =>
      expect(getByText(`${firstName} ${lastName}`)).toBeInTheDocument(),
    );
  });

  it('should route to the individual home result page when clicking the button', async () => {
    const listIndex = 1; // Index of first record with submitted answers
    const URL = '/classgroups/23234/lessons/123';
    const { getAllByText, userEvent, router } = await renderPage(URL);

    const pushSpy = jest.spyOn(router, 'push');

    const activeHomeLink = getAllByText('home')[listIndex];
    userEvent.click(activeHomeLink);
    expect(pushSpy).toHaveBeenCalledTimes(1);
    expect(
      pushSpy,
    ).toHaveBeenLastCalledWith(
      `${URL}/students/${listIndex}?viewMode=HOME`,
      `${URL}/students/${listIndex}?viewMode=HOME`,
      { shallow: undefined },
    );
  });

  it('should route to the individual class result page when clicking the button', async () => {
    const listIndex = 1; // Index of first record with submitted answers
    const URL = '/classgroups/23234/lessons/123';
    const { getAllByText, userEvent, router } = await renderPage(URL);

    const pushSpy = jest.spyOn(router, 'push');

    const activeClassLink = getAllByText('class')[listIndex];
    userEvent.click(activeClassLink);
    expect(pushSpy).toHaveBeenCalledTimes(1);
    expect(
      pushSpy,
    ).toHaveBeenLastCalledWith(
      `${URL}/students/${listIndex}?viewMode=CLASS`,
      `${URL}/students/${listIndex}?viewMode=CLASS`,
      { shallow: undefined },
    );
  });

  it('should not route to the individual class result page when clicking the button', async () => {
    const listIndex = 0; // Index of first record with NO submitted answers
    const { getAllByText, userEvent, router } = await renderPage(
      '/classgroups/23234/lessons/123',
    );

    const pushSpy = jest.spyOn(router, 'push');

    const activeClassLink = getAllByText('class')[listIndex];
    userEvent.click(activeClassLink);
    expect(pushSpy).toHaveBeenCalledTimes(0);
  });

  it('should not route to the individual home result page when clicking the button', async () => {
    const listIndex = 0; // Index of first record with NO submitted answers
    const { getAllByText, userEvent, router } = await renderPage(
      '/classgroups/23234/lessons/123',
    );

    const pushSpy = jest.spyOn(router, 'push');

    const activeClassLink = getAllByText('home')[listIndex];
    userEvent.click(activeClassLink);
    expect(pushSpy).toHaveBeenCalledTimes(0);
  });
});
