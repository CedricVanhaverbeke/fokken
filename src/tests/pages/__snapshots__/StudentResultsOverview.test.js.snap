// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentResultsOverview should render the results page when navigating to the URL 1`] = `
<div>
  <div
    class="bg-gray-100 w-full"
  >
    <div
      class="w-full flex flex-col pt-8"
    >
      <div
        class="mb-4 rounded w-full max-w-5/6 self-center"
      >
        <div
          class="flex flex-col"
        >
          <h1
            class="text-4xl font-bold leading-none"
          >
            Resultaten
          </h1>
          <span
            class="text-xl font-medium text-gray-600"
          >
            Class 1A - (a)Sociale media
          </span>
        </div>
      </div>
      <hr
        class="border-gray-300"
      />
    </div>
  </div>
  <div
    class="w-full max-w-5/6 mt-8 mb-4 flex flex-1"
  >
    <table
      class="overflow-x-auto w-full border border-gray-200 rounded-md"
      role="table"
    >
      <thead
        class="rounded-t-md"
      >
        <tr
          class="uppercase text-xs leading-4 tracking-wide text-left rounded-t text-gray-600 bg-gray-200"
          role="row"
        >
          <th
            class="p-2 font-normal"
            colspan="1"
            role="columnheader"
          >
            Name
          </th>
          <th
            class="p-2 font-normal"
            colspan="1"
            role="columnheader"
          >
            Submitted At
          </th>
          <th
            class="p-2 font-normal"
            colspan="1"
            role="columnheader"
          >
            Results
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr
          class="border-b border-gray-200"
          role="row"
        >
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex gap-x-4 items-center"
              title="Elenor_Jacobs35"
            >
              <div
                class="w-10 h-10 flex-shrink-0 text-white bg-gray-400 text-sm rounded-full leading-none flex items-center justify-center select-none"
              >
                <span>
                  C
                  S
                </span>
              </div>
              Carolyne Schamberger
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-start"
            />
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-center"
            >
              <span
                class="cursor-none text-blue-200 cursor-not-allowed"
              >
                home
              </span>
              <div
                class="w-px h-4 ml-3 mr-3 bg-gray-300"
              />
              <span
                class="cursor-none text-blue-200 cursor-not-allowed"
              >
                class
              </span>
            </div>
          </td>
        </tr>
        <tr
          class="border-b border-gray-200"
          role="row"
        >
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex gap-x-4 items-center"
              title="Melany.Lehner"
            >
              <div
                class="w-10 h-10 flex-shrink-0 text-white bg-gray-400 text-sm rounded-full leading-none flex items-center justify-center select-none"
              >
                <span>
                  J
                  G
                </span>
              </div>
              Jaron Gottlieb
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-start"
            >
              <div
                class="text-sm bg-green-200 text-green-800 rounded-sm px-2 flex"
              >
                09/07/2020
                 
                04:57
              </div>
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-center"
            >
              <span
                class="cursor-pointer text-blue-700 "
              >
                home
              </span>
              <div
                class="w-px h-4 ml-3 mr-3 bg-gray-300"
              />
              <span
                class="cursor-pointer text-blue-700 "
              >
                class
              </span>
            </div>
          </td>
        </tr>
        <tr
          class="border-b border-gray-200"
          role="row"
        >
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex gap-x-4 items-center"
              title="Carmen45"
            >
              <div
                class="w-10 h-10 flex-shrink-0 text-white bg-gray-400 text-sm rounded-full leading-none flex items-center justify-center select-none"
              >
                <span>
                  A
                  S
                </span>
              </div>
              Adelle Schoen
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-start"
            >
              <div
                class="text-sm bg-green-200 text-green-800 rounded-sm px-2 flex"
              >
                09/07/2020
                 
                04:57
              </div>
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-center"
            >
              <span
                class="cursor-pointer text-blue-700 "
              >
                home
              </span>
              <div
                class="w-px h-4 ml-3 mr-3 bg-gray-300"
              />
              <span
                class="cursor-pointer text-blue-700 "
              >
                class
              </span>
            </div>
          </td>
        </tr>
        <tr
          class="border-b border-gray-200"
          role="row"
        >
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex gap-x-4 items-center"
              title="Jett53"
            >
              <div
                class="w-10 h-10 flex-shrink-0 text-white bg-gray-400 text-sm rounded-full leading-none flex items-center justify-center select-none"
              >
                <span>
                  J
                  C
                </span>
              </div>
              Jaden Cruickshank
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-start"
            >
              <div
                class="text-sm bg-green-200 text-green-800 rounded-sm px-2 flex"
              >
                09/07/2020
                 
                04:57
              </div>
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-center"
            >
              <span
                class="cursor-pointer text-blue-700 "
              >
                home
              </span>
              <div
                class="w-px h-4 ml-3 mr-3 bg-gray-300"
              />
              <span
                class="cursor-pointer text-blue-700 "
              >
                class
              </span>
            </div>
          </td>
        </tr>
        <tr
          class="border-b border-gray-200"
          role="row"
        >
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex gap-x-4 items-center"
              title="Einar.Powlowski78"
            >
              <div
                class="w-10 h-10 flex-shrink-0 text-white bg-gray-400 text-sm rounded-full leading-none flex items-center justify-center select-none"
              >
                <span>
                  M
                  H
                </span>
              </div>
              Mattie Hahn
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-start"
            />
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-center"
            >
              <span
                class="cursor-none text-blue-200 cursor-not-allowed"
              >
                home
              </span>
              <div
                class="w-px h-4 ml-3 mr-3 bg-gray-300"
              />
              <span
                class="cursor-none text-blue-200 cursor-not-allowed"
              >
                class
              </span>
            </div>
          </td>
        </tr>
        <tr
          class="border-b border-gray-200"
          role="row"
        >
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex gap-x-4 items-center"
              title="Clara_Collier61"
            >
              <div
                class="w-10 h-10 flex-shrink-0 text-white bg-gray-400 text-sm rounded-full leading-none flex items-center justify-center select-none"
              >
                <span>
                  R
                  M
                </span>
              </div>
              River Moen
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-start"
            >
              <div
                class="text-sm bg-green-200 text-green-800 rounded-sm px-2 flex"
              >
                09/07/2020
                 
                04:57
              </div>
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-center"
            >
              <span
                class="cursor-pointer text-blue-700 "
              >
                home
              </span>
              <div
                class="w-px h-4 ml-3 mr-3 bg-gray-300"
              />
              <span
                class="cursor-pointer text-blue-700 "
              >
                class
              </span>
            </div>
          </td>
        </tr>
        <tr
          class="border-b border-gray-200"
          role="row"
        >
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex gap-x-4 items-center"
              title="Pauline_Lueilwitz"
            >
              <div
                class="w-10 h-10 flex-shrink-0 text-white bg-gray-400 text-sm rounded-full leading-none flex items-center justify-center select-none"
              >
                <span>
                  Y
                  W
                </span>
              </div>
              Yadira Wisozk
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-start"
            >
              <div
                class="text-sm bg-green-200 text-green-800 rounded-sm px-2 flex"
              >
                09/07/2020
                 
                04:57
              </div>
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-center"
            >
              <span
                class="cursor-pointer text-blue-700 "
              >
                home
              </span>
              <div
                class="w-px h-4 ml-3 mr-3 bg-gray-300"
              />
              <span
                class="cursor-pointer text-blue-700 "
              >
                class
              </span>
            </div>
          </td>
        </tr>
        <tr
          class="border-b border-gray-200"
          role="row"
        >
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex gap-x-4 items-center"
              title="Trycia_Green"
            >
              <div
                class="w-10 h-10 flex-shrink-0 text-white bg-gray-400 text-sm rounded-full leading-none flex items-center justify-center select-none"
              >
                <span>
                  D
                  O
                </span>
              </div>
              Darryl O'Connell
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-start"
            >
              <div
                class="text-sm bg-green-200 text-green-800 rounded-sm px-2 flex"
              >
                09/07/2020
                 
                04:57
              </div>
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-center"
            >
              <span
                class="cursor-pointer text-blue-700 "
              >
                home
              </span>
              <div
                class="w-px h-4 ml-3 mr-3 bg-gray-300"
              />
              <span
                class="cursor-pointer text-blue-700 "
              >
                class
              </span>
            </div>
          </td>
        </tr>
        <tr
          class="border-b border-gray-200"
          role="row"
        >
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex gap-x-4 items-center"
              title="Bianka_Lindgren64"
            >
              <div
                class="w-10 h-10 flex-shrink-0 text-white bg-gray-400 text-sm rounded-full leading-none flex items-center justify-center select-none"
              >
                <span>
                  S
                  S
                </span>
              </div>
              Shakira Simonis
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-start"
            />
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-center"
            >
              <span
                class="cursor-none text-blue-200 cursor-not-allowed"
              >
                home
              </span>
              <div
                class="w-px h-4 ml-3 mr-3 bg-gray-300"
              />
              <span
                class="cursor-none text-blue-200 cursor-not-allowed"
              >
                class
              </span>
            </div>
          </td>
        </tr>
        <tr
          class="border-b border-gray-200"
          role="row"
        >
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex gap-x-4 items-center"
              title="Larue_Williamson18"
            >
              <div
                class="w-10 h-10 flex-shrink-0 text-white bg-gray-400 text-sm rounded-full leading-none flex items-center justify-center select-none"
              >
                <span>
                  J
                  W
                </span>
              </div>
              Julius Will
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-start"
            >
              <div
                class="text-sm bg-green-200 text-green-800 rounded-sm px-2 flex"
              >
                09/07/2020
                 
                04:57
              </div>
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-center"
            >
              <span
                class="cursor-pointer text-blue-700 "
              >
                home
              </span>
              <div
                class="w-px h-4 ml-3 mr-3 bg-gray-300"
              />
              <span
                class="cursor-pointer text-blue-700 "
              >
                class
              </span>
            </div>
          </td>
        </tr>
        <tr
          class="border-b border-gray-200"
          role="row"
        >
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex gap-x-4 items-center"
              title="Jonathon54"
            >
              <div
                class="w-10 h-10 flex-shrink-0 text-white bg-gray-400 text-sm rounded-full leading-none flex items-center justify-center select-none"
              >
                <span>
                  L
                  C
                </span>
              </div>
              Laverna Cormier
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-start"
            >
              <div
                class="text-sm bg-green-200 text-green-800 rounded-sm px-2 flex"
              >
                09/07/2020
                 
                04:57
              </div>
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-center"
            >
              <span
                class="cursor-pointer text-blue-700 "
              >
                home
              </span>
              <div
                class="w-px h-4 ml-3 mr-3 bg-gray-300"
              />
              <span
                class="cursor-pointer text-blue-700 "
              >
                class
              </span>
            </div>
          </td>
        </tr>
        <tr
          class="border-b border-gray-200"
          role="row"
        >
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex gap-x-4 items-center"
              title="Maurine.Romaguera95"
            >
              <div
                class="w-10 h-10 flex-shrink-0 text-white bg-gray-400 text-sm rounded-full leading-none flex items-center justify-center select-none"
              >
                <span>
                  M
                  L
                </span>
              </div>
              Mitchel Ledner
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-start"
            >
              <div
                class="text-sm bg-green-200 text-green-800 rounded-sm px-2 flex"
              >
                09/07/2020
                 
                04:57
              </div>
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-center"
            >
              <span
                class="cursor-pointer text-blue-700 "
              >
                home
              </span>
              <div
                class="w-px h-4 ml-3 mr-3 bg-gray-300"
              />
              <span
                class="cursor-pointer text-blue-700 "
              >
                class
              </span>
            </div>
          </td>
        </tr>
        <tr
          class="border-b border-gray-200"
          role="row"
        >
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex gap-x-4 items-center"
              title="Kaci_Feest"
            >
              <div
                class="w-10 h-10 flex-shrink-0 text-white bg-gray-400 text-sm rounded-full leading-none flex items-center justify-center select-none"
              >
                <span>
                  K
                  D
                </span>
              </div>
              Katheryn Dooley
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-start"
            />
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-center"
            >
              <span
                class="cursor-none text-blue-200 cursor-not-allowed"
              >
                home
              </span>
              <div
                class="w-px h-4 ml-3 mr-3 bg-gray-300"
              />
              <span
                class="cursor-none text-blue-200 cursor-not-allowed"
              >
                class
              </span>
            </div>
          </td>
        </tr>
        <tr
          class="border-b border-gray-200"
          role="row"
        >
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex gap-x-4 items-center"
              title="Darrel82"
            >
              <div
                class="w-10 h-10 flex-shrink-0 text-white bg-gray-400 text-sm rounded-full leading-none flex items-center justify-center select-none"
              >
                <span>
                  E
                  B
                </span>
              </div>
              Edmund Bailey
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-start"
            >
              <div
                class="text-sm bg-green-200 text-green-800 rounded-sm px-2 flex"
              >
                09/07/2020
                 
                04:57
              </div>
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-center"
            >
              <span
                class="cursor-pointer text-blue-700 "
              >
                home
              </span>
              <div
                class="w-px h-4 ml-3 mr-3 bg-gray-300"
              />
              <span
                class="cursor-pointer text-blue-700 "
              >
                class
              </span>
            </div>
          </td>
        </tr>
        <tr
          class="border-b border-gray-200"
          role="row"
        >
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex gap-x-4 items-center"
              title="Maryjane_Klein"
            >
              <div
                class="w-10 h-10 flex-shrink-0 text-white bg-gray-400 text-sm rounded-full leading-none flex items-center justify-center select-none"
              >
                <span>
                  D
                  J
                </span>
              </div>
              Dimitri Jenkins
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-start"
            >
              <div
                class="text-sm bg-green-200 text-green-800 rounded-sm px-2 flex"
              >
                09/07/2020
                 
                04:57
              </div>
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-center"
            >
              <span
                class="cursor-pointer text-blue-700 "
              >
                home
              </span>
              <div
                class="w-px h-4 ml-3 mr-3 bg-gray-300"
              />
              <span
                class="cursor-pointer text-blue-700 "
              >
                class
              </span>
            </div>
          </td>
        </tr>
        <tr
          class="border-b border-gray-200"
          role="row"
        >
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex gap-x-4 items-center"
              title="Jeromy_Graham"
            >
              <div
                class="w-10 h-10 flex-shrink-0 text-white bg-gray-400 text-sm rounded-full leading-none flex items-center justify-center select-none"
              >
                <span>
                  R
                  M
                </span>
              </div>
              Rudy Metz
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-start"
            >
              <div
                class="text-sm bg-green-200 text-green-800 rounded-sm px-2 flex"
              >
                09/07/2020
                 
                04:57
              </div>
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-center"
            >
              <span
                class="cursor-pointer text-blue-700 "
              >
                home
              </span>
              <div
                class="w-px h-4 ml-3 mr-3 bg-gray-300"
              />
              <span
                class="cursor-pointer text-blue-700 "
              >
                class
              </span>
            </div>
          </td>
        </tr>
        <tr
          class="border-b border-gray-200"
          role="row"
        >
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex gap-x-4 items-center"
              title="Polly.Koelpin"
            >
              <div
                class="w-10 h-10 flex-shrink-0 text-white bg-gray-400 text-sm rounded-full leading-none flex items-center justify-center select-none"
              >
                <span>
                  R
                  B
                </span>
              </div>
              Rowena Bahringer
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-start"
            />
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-center"
            >
              <span
                class="cursor-none text-blue-200 cursor-not-allowed"
              >
                home
              </span>
              <div
                class="w-px h-4 ml-3 mr-3 bg-gray-300"
              />
              <span
                class="cursor-none text-blue-200 cursor-not-allowed"
              >
                class
              </span>
            </div>
          </td>
        </tr>
        <tr
          class="border-b border-gray-200"
          role="row"
        >
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex gap-x-4 items-center"
              title="Dulce_Vandervort"
            >
              <div
                class="w-10 h-10 flex-shrink-0 text-white bg-gray-400 text-sm rounded-full leading-none flex items-center justify-center select-none"
              >
                <span>
                  A
                  M
                </span>
              </div>
              Alana Mohr
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-start"
            >
              <div
                class="text-sm bg-green-200 text-green-800 rounded-sm px-2 flex"
              >
                09/07/2020
                 
                04:57
              </div>
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-center"
            >
              <span
                class="cursor-pointer text-blue-700 "
              >
                home
              </span>
              <div
                class="w-px h-4 ml-3 mr-3 bg-gray-300"
              />
              <span
                class="cursor-pointer text-blue-700 "
              >
                class
              </span>
            </div>
          </td>
        </tr>
        <tr
          class="border-b border-gray-200"
          role="row"
        >
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex gap-x-4 items-center"
              title="Ashley38"
            >
              <div
                class="w-10 h-10 flex-shrink-0 text-white bg-gray-400 text-sm rounded-full leading-none flex items-center justify-center select-none"
              >
                <span>
                  C
                  K
                </span>
              </div>
              Clint Klocko
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-start"
            >
              <div
                class="text-sm bg-green-200 text-green-800 rounded-sm px-2 flex"
              >
                09/07/2020
                 
                04:57
              </div>
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-center"
            >
              <span
                class="cursor-pointer text-blue-700 "
              >
                home
              </span>
              <div
                class="w-px h-4 ml-3 mr-3 bg-gray-300"
              />
              <span
                class="cursor-pointer text-blue-700 "
              >
                class
              </span>
            </div>
          </td>
        </tr>
        <tr
          class="border-b border-gray-200"
          role="row"
        >
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex gap-x-4 items-center"
              title="Ulises.Effertz"
            >
              <div
                class="w-10 h-10 flex-shrink-0 text-white bg-gray-400 text-sm rounded-full leading-none flex items-center justify-center select-none"
              >
                <span>
                  G
                  B
                </span>
              </div>
              Gwendolyn Botsford
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-start"
            >
              <div
                class="text-sm bg-green-200 text-green-800 rounded-sm px-2 flex"
              >
                09/07/2020
                 
                04:57
              </div>
            </div>
          </td>
          <td
            class="p-2 whitespace-no-wrap"
            role="cell"
          >
            <div
              class="flex items-center"
            >
              <span
                class="cursor-pointer text-blue-700 "
              >
                home
              </span>
              <div
                class="w-px h-4 ml-3 mr-3 bg-gray-300"
              />
              <span
                class="cursor-pointer text-blue-700 "
              >
                class
              </span>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
